solo/fido2/__init__.py

```diff
@@ -56,3 +57,27 @@ def Read(self):
             except TypeError:
                 msg[i] = v
         return msg
+
+# Linux fix for serial number:
+if sys.platform.startswith("linux"):
+    import fido2._pyu2f.linux
+
+    # Adjust function ParseUevent from python-fido2
+    def newParseUevent(uevent, desc):
+        lines = uevent.split(b'\n')
+        for line in lines:
+            line = line.strip()
+            if not line:
+                continue
+            k, v = line.split(b'=')
+            if k == b'HID_NAME':
+                desc.product_string = v.decode('utf8')
+            elif k == b'HID_ID':
+                _, vid, pid = v.split(b':')
+                desc.vendor_id = int(vid, 16)
+                desc.product_id = int(pid, 16)
+            elif k == b"HID_UNIQ":
+                desc.serial_number = v.decode("utf8")
+
+    fido2._pyu2f.linux.ParseUevent = newParseUevent
+
```

Comment:
This should stay together with the other patches (and ideally support Win/macOS too, but that's optional).

Reply:
My reasoning for moving it was that it's not a CLI patch anymore but a full functionality patch (in that it affects both normal python usage and CLI functions). Happy to move it back if you want.

Regarding Windows/MacOs I didn't make changes about them for two reasons. Wanted a smaller PR to get initial feedback and see if the approach was ok from your side. Moreover I don't have Windows/MacOs machines to test any changes.